test_util/test_upgrade_vpc.py

```diff
@@ -173,35 +268,11 @@ def main():
 
     cluster_api.wait_for_dcos()
 
-    # Deploy an app, and make sure deployments are completely done.
-    cluster_api.marathon.deploy_app(get_test_app())
-
-    cluster_api.marathon.ensure_deployments_complete()
-
-    task_info_before_upgrade = get_task_info(cluster_api.marathon.get('v2/apps').json(),
-                                             cluster_api.marathon.get('v2/tasks').json())
-
-    assert task_info_before_upgrade is not None, "Unable to get task details of the cluster."
-    assert task_info_before_upgrade.state == "TASK_RUNNING", "Task is not in the running state."
-
     with cluster.ssher.tunnel(cluster.bootstrap_host) as bootstrap_host_tunnel:
         bootstrap_host_tunnel.remote_cmd(['sudo', 'rm', '-rf', cluster.ssher.home_dir + '/*'])
 
-    test_util.cluster.upgrade_dcos(cluster, installer_url, add_config_path=config_yaml_override_upgrade)
-
-    task_info_after_upgrade = get_task_info(cluster_api.marathon.get('v2/apps').json(),
-                                            cluster_api.marathon.get('v2/tasks').json())
-
-    assert task_info_after_upgrade is not None, "Unable to get the tasks details of the cluster."
-    assert task_info_after_upgrade.state == "TASK_RUNNING", "Task is not in the running state."
-
-    assert task_info_before_upgrade.id == task_info_after_upgrade.id, \
```

Comment:
`/marathon/v2/app/<app_id>/tasks` only returns the running tasks for that app. So we can assume the tasks we get from that endpoint are running.

https://github.com/dcos/dcos/pull/1105/commits/ecb92e25c5b4e907ec4605af491e0dc47b715499 redefines the healthcheck app such that a single healthcheck failure causes Marathon to kill the task, which we'll then catch here. (That's explained in the commit message as well as a comment next to the healthcheck app definition.) My theory here is that this function's complexity will grow pretty quickly if every new app needs to be verified on exit in its own particular way. If we can define long-running workloads such that failure is reflected as task failure, then we can verify all those workloads at once just by comparing their running task IDs at start and end.

This does drop the explicit check that Marathon executed at least one health check post-upgrade. I don't think that's appropriate to test here -- that's testing specific Marathon behavior, when this function should be focused on testing whether workloads were disrupted between entry and exit. In other words, this function should assume that Marathon works. If we want to verify that the Marathon version we upgraded to correctly executes health checks, we should capture that behavior in its own integration test, which will then get run against the upgraded cluster. And as a bonus, that test would also get run in our other integration tests.